src/main/java/net/starschema/clouddb/cmdlineverification/Oauth2Bigquery.java

```diff
@@ -240,8 +232,49 @@ public static Bigquery authorizeviaservice(String serviceaccountemail,
             PrivateKey pk = getPrivateKeyFromCredentials(keypath, password);
             builder = builder.setServiceAccountPrivateKey(pk);
         }
+        return builder.build();
+    }
+
+    /**
+     * This function gives back an built GoogleCredential Object from a json keyfile
+     *
+     * @param keypath
+     * @return Built GoogleCredential via and keypath
+     * @throws GeneralSecurityException
+     * @throws IOException
+     */
+    private static GoogleCredential createJsonCredential(String keypath) throws GeneralSecurityException, IOException {
+        logger.debug("Authorizing with service account.");
+        // For .json load the key via credential.fromStream
+        File jsonKey = new File(keypath);
+        InputStream inputStream = new FileInputStream(jsonKey);
+        return GoogleCredential.fromStream(inputStream, CmdlineUtils.getHttpTransport(), CmdlineUtils.getJsonFactory()).createScoped(GenerateScopes());
+    }
+
+    /**
+     * This function gives back an Authorized Bigquery Client It uses a service
+     * account, which doesn't need user interaction for connect
+     *
+     * @param serviceaccountemail
+     * @param keypath
+     * @return Authorized Bigquery Client via serviceaccount e-mail and keypath
+     * @throws GeneralSecurityException
+     * @throws IOException
+     */
+    public static Bigquery authorizeviaservice(String serviceaccountemail,
+                                               String keypath,
+                                               String password,
+                                               String userAgent,
+                                               Integer connectTimeout,
+                                               Integer readTimeout) throws GeneralSecurityException, IOException {
+        GoogleCredential credential;
+        // Determine which keyfile we are trying to authenticate with.
+        if (Pattern.matches(".*\\.json$", keypath)) {
+            credential = Oauth2Bigquery.createJsonCredential(keypath);
+        } else {
+            credential = Oauth2Bigquery.createP12Credential(serviceaccountemail, keypath, password);
+        }
```

Comment:
Yep this was along the lines of what I was thinking to keep the interface the same.